scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,474 @@
+    // Maps element name to its index in the data vector.
+    schema: &'schema CDCRowSchema,
+}
+
+impl CDCRow<'_> {
+    pub fn from_row(row: Row, schema: &CDCRowSchema) -> CDCRow {
+        // If cdc read was successful, these default values will not be used.
+        let mut stream_id = vec![];
+        let mut time = uuid::Uuid::default();
+        let mut batch_seq_no = i32::MAX;
+        let mut end_of_batch = false;
+        let mut operation = OperationType::PreImage;
+        let mut ttl = None;
+
+        let data_count =
+            schema.mapping.len() + schema.deleted_mapping.len() + schema.deleted_el_mapping.len();
+        let mut data: Vec<Option<CqlValue>> = Vec::with_capacity(data_count);
+
+        for (i, column) in row.columns.into_iter().enumerate() {
+            if i == schema.stream_id {
+                stream_id = column.unwrap().into_blob().unwrap();
+            } else if i == schema.time {
+                time = column.unwrap().as_uuid().unwrap();
+            } else if i == schema.batch_seq_no {
+                batch_seq_no = column.unwrap().as_int().unwrap();
+            } else if i == schema.end_of_batch {
+                end_of_batch = column.unwrap().as_boolean().unwrap()
+            } else if i == schema.operation {
+                operation = OperationType::try_from(column.unwrap().as_tinyint().unwrap()).unwrap();
+            } else if i == schema.ttl {
+                ttl = column.map(|ttl| ttl.as_bigint().unwrap());
+            } else {
+                data.push(column);
+            }
+        }
+
+        CDCRow {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            data,
+            schema,
+        }
+    }
+
+    /// Allows to get a value from the column that corresponds to the logged table.
+    /// Returns None if there is no column with such name.
+    /// Returns Some(None) if such column exists, but its value is null.
+    /// Otherwise returns Some(&Some(x)) where x is the value in that column.
+    pub fn get_value(&self, name: &str) -> Option<&Option<CqlValue>> {
+        self.schema.mapping.get(name).map(|id| &self.data[*id])
+    }
+
+    /// Allows to get info if a value was deleted in this operation.
+    /// Returns None if there is no column with such name.
+    pub fn is_value_deleted(&self, name: &str) -> Option<bool> {
+        self.schema
+            .deleted_mapping
+            .get(name)
+            .map(|id| self.data[*id].is_some())
+    }
+
+    /// Allows to get deleted elements from a collection.
+    /// Returns None if there is no collection column with such name.
+    /// Returns Some(None) if there is such collection, but nothing was deleted from it.
+    /// Otherwise returns Some(Some(x)) where x is a reference to vector containing deleted values.
+    pub fn get_deleted_elements(&self, name: &str) -> Option<Option<&Vec<CqlValue>>> {
```

Comment:
We can simplify a bit here, similarly to `is_value_deleted`. I don't think that the set of deleted elements can be empty, so we can consider null to be an empty Vec in order to remove one layer of Option. Additionally, you should change `&Vec<CqlValue>` to `&[CqlValue]` - always prefer slices to vec references.

Also, take a look at my summary comment of this review - I included more suggestions which will affect the final shape of this function.